ssz/sedes/serializable.py

```diff
@@ -169,21 +170,41 @@ def copy(self, *args, **kwargs):
         ).difference(
             self._meta.field_names[:len(args)]
         )
-        unchanged_kwargs = {
-            key: copy.deepcopy(value)
-            for key, value
-            in self.as_dict().items()
-            if key in missing_overrides
-        }
+        unchanged_kwargs = {}
+        for key, value in self.as_dict().items():
+            if key in missing_overrides:
+                is_immutable_field = (
+                    isinstance(value, BASE_TYPES) or
+                    (
+                        isinstance(value, tuple) and
+                        len(value) > 0 and
+                        isinstance(value[0], BASE_TYPES)
+                    )
+                )
+                if is_immutable_field:
+                    unchanged_kwargs[key] = value
+                else:
+                    unchanged_kwargs[key] = copy.deepcopy(value)
+
         combined_kwargs = dict(**unchanged_kwargs, **kwargs)
         all_kwargs = merge_args_to_kwargs(args, combined_kwargs, self._meta.field_names)
         return type(self)(**all_kwargs)
 
     def __copy__(self):
         return self.copy()
 
-    def __deepcopy__(self, *args):
-        return self.copy()
+    def __deepcopy__(self, memodict=None):
+        if memodict is None:
+            memodict = {}
+
+        cls = self.__class__
+        result = cls.__new__(cls)
```

Comment:
this looks strange to me. Why can't we do something like this?

```Py
result = cls(**{
    key: copy.deepcopy(value) for key, value in self.to_dict().items()
})
```

(or maybe, while we're at it, even add a classmethod `from_dict`)? Or is calling `__init__` too expensive performance-wise?